### DISCUSSION TITLE
How do i listen to changes in database and update my ui

### DISCUSSION THREAD

**Discussion by kcibdev** [supabase/supabase#2783] [Questions] [answered]
Am using supabase with dart/flutter and i don't really know know how to listen to changes in the database like check if there is changes from a specific userId or a specific column in the database or if someone sent a message to me using my id then my UI 
 should be updated immediately just like how firebase does by using `QuerySnapshots` with `streambuilder` is that possible with supabase at the moment

**Comment by dshukertjr** [comment, marked as answer]
Hi @BrightCode1!

You can listen to realtime updates of your Supabase database by using [`realtime` feature](https://github.com/supabase/supabase-dart#realtime).

```dart
// Set up a listener to listen to changes in `countries` table
  final subscription = await supabase
  .from('countries')
  .on(SupabaseEventTypes.all, (payload) {
    // Do something when there is an update
  })
  .subscribe();

// remember to remove subscription when you're done
supabase.removeSubscription(subscription);
```

To listen to rows where userId is set to a specific userId, you can do something like this:
```dart
// Set up a listener to listen to changes in `countries` table
  final subscription = await supabase
  .from('posts:user_id=eq.${userId}')
  .on(SupabaseEventTypes.all, (payload) {
    // Do something when there is an update
  })
  .subscribe();
```

You can read more about how to use `realtime` on the official documentation. Although this doc references the JavaScript library, most of the syntax is the same. 
https://supabase.io/docs/reference/javascript/subscribe
